README.md

```diff
@@ -1,3 +1,137 @@
-# nextflow-lamin
+# nf-lamin plugin
 
-This repository contains examples on how to use Lamin together with Nextflow to track workflow execution and data lineage.
+A Nextflow plugin that integrates LaminDB data provenance into Nextflow
+workflows.
+
+## Installation
+
+To install a development build of the plugin, run the following command:
+
+```bash
+make install
+```
+
+To uninstall the plugin, run the following command:
+
+```bash
+make uninstall
+```
+
+## Usage
+
+To use the plugin in a Nextflow workflow, create a Nextflow config
+`nextflow.config` that specifies which LaminDB instance to use and an
+API key:
+
+```groovy
+lamin {
+  instance = "laminlabs/lamindata"
+  api_key = System.getenv("LAMIN_API_KEY")
+}
+```
+
+You can now use the plugin by adding the following option to the
+`nextflow run` command:
+
+```bash
+-plugins nf-lamin@0.0.1
+```
+
+> [!TIP]
+>
+> You can fetch your Lamin API key from your [Lamin Hub account
+> settings](https://lamin.ai/settings).
+
+### Example with scrnaseq workflow:
+
+```bash
+nextflow run nf-core/scrnaseq \
+  -latest -resume \
+  -plugins nf-lamin@0.0.1 \
+  -r 2.7.1 \
+  -profile docker \
+  --input https://github.com/nf-core/test-datasets/raw/scrnaseq/samplesheet-2-0.csv \
+  --fasta https://github.com/nf-core/test-datasets/raw/scrnaseq/reference/GRCm38.p6.genome.chr19.fa \
+  --gtf https://github.com/nf-core/test-datasets/raw/scrnaseq/reference/gencode.vM19.annotation.chr19.gtf \
+  --protocol 10XV2 \
+  --skip_emptydrops \
+  --outdir output/scrnaseq
+```
+
+### Example with Viash Hub workflow:
+
+```bash
+nextflow run https://packages.viash-hub.com/vsh/toolbox.git \
+  -revision v0.1.0 \
+  -main-script target/nextflow/bgzip/main.nf \
+  -profile docker \
+  -latest \
+  -plugins nf-lamin@0.0.1 \
+  --input https://github.com/nf-core/test-datasets/raw/scrnaseq/samplesheet-2-0.csv \
+  --output output.gz \
+  --publish_dir gs://di-temporary/scratch/temp-nf-lamin
```

Comment:
Use a local directory so it works for people without storage permissions

```suggestion
  --publish_dir output/bgzip
```

Reply:
I wanted to demonstrate what happens with remote buckets, because publishing files locally would require you to also upload the files in order to get the artifacts to work properly...

Reply:
Good point

Reply:
Done!